libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LGExtensions.cs

```diff
@@ -9,57 +9,48 @@
 
 namespace Microsoft.Bot.Builder.Dialogs.Adaptive
 {
-    /// <summary>
-    /// LG middleware.
-    /// </summary>
-    public static class LGAdapterExtensions
+    public static class LGExtensions
     {
         private static Dictionary<ResourceExplorer, LanguageGeneratorManager> languageGeneratorManagers = new Dictionary<ResourceExplorer, LanguageGeneratorManager>();
 
         /// <summary>
         /// Register default LG file as language generation.
         /// </summary>
-        /// <param name="botAdapter">The <see cref="BotAdapter"/> to add services to.</param>
-        /// <param name="resourceExplorer">resource explorer to use for .lg based resources.</param>
+        /// <param name="dialogManager">The <see cref="BotAdapter"/> to add services to.</param>
         /// <param name="defaultLg">Default LG Resource Id (default: main.lg).</param>
         /// <returns>The BotAdapter.</returns>
-        public static BotAdapter UseLanguageGeneration(
-            this BotAdapter botAdapter,
-            ResourceExplorer resourceExplorer = null,
+        public static DialogManager UseLanguageGeneration(
+            this DialogManager dialogManager,
             string defaultLg = null)
         {
```

Comment:
How does this relate to replacing the generator?  Or does it?

Reply:
this is how you do it.  We just moved it from botAdapter => DialogManager

---
In reply to: [376542764](https://github.com/microsoft/botbuilder-dotnet/pull/3350#discussion_r376542764) [](ancestors = 376542764)